pkg/apis/controlplane/sets.go

```diff
@@ -213,3 +217,36 @@ func (s GroupMemberSet) Items() []*GroupMember {
 	}
 	return res
 }
+
+// UInt32 maintains a set of types uint32, implemented via
```

Comment:
if we switch to `int32` for tier priority, we can use `sets.Int32` directly, which is also a plus and helps keep the diff smaller